two-factor authentication

gr2m opened the pull request:
closes #669

mattfysh commented:
@gr2m - trying to use this snippet, and it keeps asking me for my 2FA code. Let me know if I'm doing something wrong.

```
octokit.authenticate({
  ...ghAuth,
  on2fa: () => new Promise(resolve => {
    rl.question('Enter your GitHub 2FA code: ', resolve)
  })
})
```

gr2m commented:
Hmm, does it end up in a loop? I thought I fixed that via https://github.com/octokit/rest.js/commit/d29ec5aca70c55a04cdccb020ac0cfaa3f256ebd 🤔 

I added a test to make sure that async on2fa works as expected: https://github.com/octokit/rest.js/pull/1213

I cannot reproduce the problem yet I’m afraid
